### ISSUE TITLE
Re-assign same value should not consider as changed/dirty attributes

### ISSUE THREAD

**Issue by mkhairi** [flexirest/flexirest#129]
### Expected behaviour

Attributes marked as dirty even though I assign same value.
```ruby
account = Account.first
account._attributes 
# {account_name: "Test", account_number: "xxxx"}
account.account_name = "Test"
account.account_number= "yyyy"
account.changed
# [:account_name, :account_number]
```

### Desired behaviour

It should return `# [:account_number]` only since account_name has the same previous value

**Comment by andyjeffries** [comment]
Thank you. Released as 1.9.15.

**Comment by AKRathore** [comment]
Hi, Greetings
Actually I am facing the same issue with `1.9.15 version`.
But the case is different now, it works fine if i try:-
```
objects = Account.all
objects.first.changed? 
#return false, which is correct
```
But it doesn't work well if I try with find method, like:-
```
object = Account.find(id: 1)
object.changed? 
#return true
object.changes 
# {:id=>[nil, 1], :account_number=>[nil, 208],.....}
```
Thanks

**Comment by AKRathore** [comment]
@andyjeffries thanks for your test case, it really helped me to confirm the issue.
And i got the reason. Actually in response, i am getting type attribute, which is the root cause for the issue, if i set the type = nil in DB, then fetch the records again then it works well.

**Comment by andyjeffries** [comment]
Weird, but glad it helped you find it. Are you using JsonAPI or something? There's nothing I can see/remember in basic Flexirest that does specific handling of a type attribute.

**Comment by AKRathore** [comment]
Actually i m integrating it with Rails, without specifying any special format. Not sure if it is the answer for your question.
Normally the type attribute used for the STI, to describe the child class.

**Comment by andyjeffries** [comment]
I don't think it's related to STI (either from the server side or the client side).

I've created a simple Rails app to try to reproduce your issue and can't - https://github.com/flexirest/issue-reproduction-129

There's a simple controller https://github.com/flexirest/issue-reproduction-129/blob/master/app/controllers/people_controller.rb that just returns JSON with an ID, Type and Name.

If you `bundle install` and run `bundle exec rails s` in one Terminal tab/window, then in a separate one run `bundle exec rails c` you should be able to do this (trimmed out some unessential lines that don't add value):

```
$ rails c
irb(main):001:0> person = RemotePerson.find(id: 1)
  Flexirest (140.3ms) RemotePerson#find
irb(main):002:0> person.changed?
=> false
irb(main):003:0> person.changes
=> {}
```

So, it works fine for me. Can you try to fork the reproduction repo and make a reproducible test case of yours where it breaks?

**Comment by AKRathore** [comment]
Hi @andyjeffries 
I just tried to reproduce the issue with your reproduction repo code, but my bad it's not related to STI. 
But seems it's somehow related to caching, because when i try the same request with adding `perform_caching false`. it works fine in my code for same records which were creating problem earlier. 
Not sure how can i check the stored response in Redis cache for that occurring response. because it seems i did something wrong at very first calls.
However i tried to paste my API `response._attributes` in PeopleController#show method response. but still no luck to reproduce. maybe the caching storing the local/manual changes of object after API call(but not sure currently)? :(
Thanks for your help, I really appreciate :1st_place_medal:

**Comment by andyjeffries** [comment]
This should be fixed as of 1.9.16. Thanks for your efforts in getting to the bottom of it.
